### PR TITLE
Fix test assertion to be called. Fix array equality test for headers.

### PR DESCRIPTION
**Pull request by mishamo**

### CODE REVIEW

File: scheduler/src/test/scala/com/sky/kms/e2e/SchedulerIntSpec.scala [hunk 1]
```diff
@@ -16,8 +16,9 @@ class SchedulerIntSpec extends SchedulerIntSpecBase {
         withSchedulerApp {
           val schedules = createSchedules(2, forTopics = List(scheduleTopic, extraScheduleTopic))
 
-          publish(schedules) andThen
-            (assertMessagesWrittenFrom(_, schedules))
+          publish(schedules)
+            .foreach(assertMessagesWrittenFrom(_, schedules))
```
Comment (mishamo) [inline scheduler/src/test/scala/com/sky/kms/e2e/SchedulerIntSpec.scala:20]: The `andThen` returns a partial function which was never actually called, so the assertion never got executed.

File: scheduler/src/test/scala/com/sky/kms/e2e/SchedulerIntSpec.scala [hunk 2]
```diff
@@ -48,7 +49,10 @@ class SchedulerIntSpec extends SchedulerIntSpecBase {
           cr.key should contain theSameElementsInOrderAs schedule.key
           cr.value should contain theSameElementsInOrderAs schedule.value.get
           cr.timestamp shouldBe time.toInstant.toEpochMilli +- Tolerance.toMillis
-          cr.headers().toArray.map(h => h.key() -> h.value()).toMap should contain theSameElementsAs schedule.headers
+          cr.headers().toArray.map(h => h.key() -> h.value().toList) should contain theSameElementsAs
+            schedule.headers.map {
+              case (k, v) => (k, v.toList)
```
Comment (mishamo) [inline scheduler/src/test/scala/com/sky/kms/e2e/SchedulerIntSpec.scala:54]: Due to the values being `Array[Byte]`, the equality test would always fail, even if they were equivalent (due to array equality basically being rubbish in scala).

File: scheduler/src/test/scala/com/sky/kms/e2e/package.scala [hunk 3]
```diff
@@ -8,7 +8,7 @@ import scala.concurrent.duration._
 
 package object e2e {
 
-  val Tolerance = 200 millis
+  val Tolerance = 400 millis
```
Comment (mishamo) [inline scheduler/src/test/scala/com/sky/kms/e2e/package.scala:11]: With the assertion enabled, I found the test to be flaky with values under 400 millis

File: scheduler/src/test/scala/com/sky/kms/e2e/SchedulerIntSpec.scala [hunk 4]
```diff
@@ -48,7 +49,10 @@ class SchedulerIntSpec extends SchedulerIntSpecBase {
           cr.key should contain theSameElementsInOrderAs schedule.key
           cr.value should contain theSameElementsInOrderAs schedule.value.get
```
Comment (mishamo) [inline scheduler/src/test/scala/com/sky/kms/e2e/SchedulerIntSpec.scala:50]: As `random[ScheduledEvent]` can generate the `value` field as `None` around 10% of the time, it causes the right hand side of this test to fail. Investigating a possible fix for this.

**Review by lacarvalho91 — APPROVED** [review]
good spot, thanks!

**Pull request merged**